### ISSUE TITLE
Musescore 3.7 does not remember favorite paths in open/save dialog.

### ISSUE THREAD

**Issue by zorba77** [Jojo-Schmitz/MuseScore#436]
Step to reproduce:
- Open MS 3.7
- CTRL+O to open a file
- On the left there are the standard paths: "/home/username/", Spartiti, bin, demos
- Drag one or more directory from the right panel to the favorite panel on the left
- Directories are shown and accessible via click
- Close and reopen MS 3.7
- user defined favorite paths are lost. Only default paths are present

The same for file save dialog (except that the demos path is absent)

OS: Ubuntu 22.04.4 LTS, Arch.: x86_64, MuseScore version (64-bit): 3.7.0.8678350200, revision: GitHub-Jojo-Schmitz-MuseScore-acbfac0

**Comment by Jojo-Schmitz** [comment]
a) does 3.6.2 behave differently?
b) does any 4.x behave differently?

**Comment by zorba77** [comment]
a) 3.6.2 works fine, it's a regression. When I open MS 3.7 it delete also old 3.6.2 favorite paths
b) 4.2.1 works fine, and it's independent from 3.X settings (favorite paths are not deleted by MS 3.7). 
4.X seems to use another file dialog, it does not seem the kde dialog, but I don't know

**Comment by Jojo-Schmitz** [comment]
OK, so a regression :-(

**Comment by zorba77** [comment]
Any news for this regression in the open/save dialog?

**Comment by Jojo-Schmitz** [comment]
As under Windows there is no such favourites panel, I'm having a hard time reproducing the issue, the file open dialog looks and works exactly thesa em in 3.6.2 and 3.7, under Windows

**Comment by Jojo-Schmitz** [comment]
And yes, 4.x after 4.2.1 doesn't use the KDE file dialog (under Linux) anymore as far as I know

**Comment by zorba77** [comment]
I suppose that also on Windows' file dialog there's a panel on the left with standard folders, but I don't remember.
How can I help you to find the problem?

**Comment by Jojo-Schmitz** [comment]
There is such a panel:
![image](https://github.com/Jojo-Schmitz/MuseScore/assets/1786669/daf17d53-17c7-4e93-8048-801ccbec145a)

**Comment by zorba77** [comment]
![dialog](https://github.com/Jojo-Schmitz/MuseScore/assets/137386945/95279f03-36a0-4818-9c42-d06a15951f38)
this is the Kubuntu open dialog.
It seems that favorite paths are stored in the file $HOME/.config/QtProject.conf
shortcuts=file:///home/ab, file:///home/ab/Documenti/MuseScore3/Spartiti, ...

**Comment by zorba77** [comment]
1. Open MS 3.6.2, open file dialog, add a new folder on the left by dragging
2. close MS 3.6.2, the file  $HOME/.config/QtProject.conf is written with a new voice in the variable "shortcuts"
3. Open MS 3.6.2, the new path is loaded and is present in the file dialog
4. Close MS 3.6.2
5. Open MS 3.7, in the file dialog there isn't the new path, so, probably, the file QtProject.conf is ignored (or something goes wrong)
6. Close MS 3.7, the file QtProject.conf is written without the new path, only default paths are restored

**Comment by zorba77** [comment]
is this useful?
https://musescore.org/en/node/75306

**Comment by zorba77** [comment]
![image](https://github.com/user-attachments/assets/4f7c9688-5e01-4073-a552-100363303a12)
this is the 3.6.2 file dialog. It's different than 3.7, see above.
The file dialog seems created in file "file.cpp", function "QStringList MuseScore::getOpenScoreNames", line 1109 and following.
See line 1143:
 // setup side bar urls
      QList<QUrl> urls = sidebarUrls();
Is it this the function that populates the side bar?
Use the native dialog may resolve the bug?

**Comment by Jojo-Schmitz** [comment]
I don't see any significalt difference between those 2 screenshots, just the section to the right.
And yes, only Linux by default uses the non-native file dialogs (which means it uses the Qt dialogs), but you can configure that in the advanced preferences

**Comment by zorba77** [comment]
Yes, because I was working on the code , sorry.
Using the native dialog also on Linux fix the problem. I suppose it's better, for non advanced users, make this option the default also on linux.
Otherwise it's necessary to investigate why the side bar isn't populated on qtdialog

> I don't see any significalt difference between those 2 screenshots, just the section to the right. And yes, only Linux by default uses, the non-native file dialogs (which means it uses the Qt dialogs), but you can configure that in the advanced preferences
> 
> You're quiting the code wrongly BTW:
> 
> ```c++
> #if defined(Q_OS_MAC) || defined(Q_OS_WIN) // use system native file dialog, Qt file dialog is very slow on Windows and Mac
>             {PREF_UI_APP_USENATIVEDIALOGS,                         new BoolPreference(true)},
> #else // don't use system native file dialog, this is causing issues on some Linuxes
>             {PREF_UI_APP_USENATIVEDIALOGS,                         new BoolPreference(false)},  // false, not true
> #endif
> ```

**Comment by zorba77** [comment]
> but you can configure that in the advanced preferences
this doesn't work, neither on MS 3.6.2 or 3.7
The native dialog appears if I set it in the file preferences.cpp, like Windows and MacOs

**Comment by Jojo-Schmitz** [comment]
It certainly works for me, on Wondows:
![image](https://github.com/user-attachments/assets/1702d3b5-727c-42b0-93ec-ab43927a64bc)

**Comment by Jojo-Schmitz** [comment]
And it doesn't remeber the favourite paths (those in the left hand part of the dialog), not in 3.6.2 nor in 3.7, so no regression there

**Comment by Jojo-Schmitz** [comment]
Mu4 doesn't use the non-native dialogs at all anymore IIRC

**Comment by zorba77** [comment]
Yes, also 3.6.2 doesn't remember favorite paths, almost on Linux. The use of Native dialogs resolve the issue, so I suppose the best solution is to make it the default choice also on Linux, like Windows and Macos, modifying the file preferences.cpp.

**Comment by Jojo-Schmitz** [comment]
But see https://musescore.org/en/node/366348

**Comment by Jojo-Schmitz** [comment]
> is this useful?
> https://musescore.org/en/node/75306

Yes it might be careful indeed
